lib/compat/wordpress-6.6/rest-api.php

```diff
@@ -29,3 +29,23 @@ function wp_api_template_access_controller( $args, $post_type ) {
 	}
 }
 add_filter( 'register_post_type_args', 'wp_api_template_access_controller', 10, 2 );
+
+function gutenberg_block_editor_preload_paths_6_6( $paths ) {
+	$paths[] = array( '/wp/v2/settings', 'OPTIONS' );
+	$paths[] = '/wp/v2/settings';
```

Comment:
The `/settings` endpoint is already preloaded, but the middle removes an item from the cache after the first request. It works fine when core data makes requests because resolvers are cached, but you can get duplicate REST API calls when it's a pure `apiFetch` call.

See related conversation - https://github.com/WordPress/gutenberg/pull/59243#issuecomment-2039253340.

Reply:
Are we making random API calls?

Reply:
Not random:

* First call to the `getSite()` selector
* The `loadSiteEntity` makes an `OPTIONS` call, fulfilled by the cache.
* Then somewhere, code calls `canUser( 'edit', 'settings' )`; this will hit the network.

Reply:
So you're saying that `getSite` and `canUser` trigger the same apiFetch call? That's weird. Can we use `getSite` with `canUser` maybe?

Reply:
The `getOrLoadEntitiesConfig` will trigger the `OPTIONS` request to load labels (used by `getEntityRecord`) for the `site` entity, and `canUser` also triggers `OPTIONS` requests. Only one of these requests is fulfilled by preloaded data.

Reply:
If we move `getOrLoadEntitiesConfig` to a separate resolver, this would be solved no?

Reply:
@Mamaduka Where do you see the `/settings OPTIONS` request preloaded? I was looking at this:

https://github.com/WordPress/wordpress-develop/blob/7159243c090e429a7d2a1fd2a9509e323f67a39d/src/wp-admin/site-editor.php#L90-L117

Reply:
Also preloaded data is stored by key, so if it was there before, adding it again wouldn't make it appear twice in the preload data?

Reply:
My bad. It looks like I've only [added preloading](https://github.com/WordPress/wordpress-develop/pull/3156/files) for the post editor.

> Also preloaded data is stored by key, so if it was there before, adding it again wouldn't make it appear twice in the preload data?

I think the paths are stored in an indexed array, so duplication is possible.

I've got an idea regarding the `getOrLoadEntitiesConfig` issue. I'll open a separate PR.

Reply:
Hm, not in my testing, `createPreloadingMiddleware` receives a object with paths as keys.

<img width="539" alt="image" src="https://github.com/WordPress/gutenberg/assets/4710635/47225795-87c3-41a1-b55b-baf39ee671fa">

Reply:
I meant the PHP side, where you can end up with duplicates, but there might be checks when actual preloading happens.

## Screenshot
![CleanShot 2024-04-25 at 15 28 45](https://github.com/WordPress/gutenberg/assets/240569/d495ea1b-366a-4471-80a2-e6ed3fba16b1)

Reply:
Right but on the client side I don't think that means the data can be used twice

Reply:
I'm not seeing any duplicates in the site editor. Is that the post editor? I'm only changing the site editor here

```
array(16) {
  [0]=>
  array(2) {
    [0]=>
    string(12) "/wp/v2/media"
    [1]=>
    string(7) "OPTIONS"
  }
  [1]=>
  string(25) "/wp/v2/types?context=view"
  [2]=>
  string(37) "/wp/v2/types/wp_template?context=edit"
  [3]=>
  string(42) "/wp/v2/types/wp_template-part?context=edit"
  [4]=>
  string(41) "/wp/v2/templates?context=edit&per_page=-1"
  [5]=>
  string(46) "/wp/v2/template-parts?context=edit&per_page=-1"
  [6]=>
  string(40) "/wp/v2/themes?context=edit&status=active"
  [7]=>
  string(39) "/wp/v2/global-styles/12464?context=edit"
  [8]=>
  string(26) "/wp/v2/global-styles/12464"
  [9]=>
  string(44) "/wp/v2/global-styles/themes/twentytwentyfour"
  [10]=>
  array(2) {
    [0]=>
    string(17) "/wp/v2/navigation"
    [1]=>
    string(7) "OPTIONS"
  }
  [11]=>
  array(2) {
    [0]=>
    string(101) "/wp/v2/navigation?context=edit&per_page=100&order=desc&orderby=date&status[0]=publish&status[1]=draft"
    [1]=>
    string(3) "GET"
  }
  [12]=>
  array(2) {
    [0]=>
    string(15) "/wp/v2/settings"
    [1]=>
    string(7) "OPTIONS"
  }
  [13]=>
  string(15) "/wp/v2/settings"
  [14]=>
  array(2) {
    [0]=>
    string(16) "/wp/v2/templates"
    [1]=>
    string(7) "OPTIONS"
  }
  [15]=>
  string(25) "/wp/v2/types?context=edit"
}
```

Reply:
> Right but on the client side I don't think that means the data can be used twice

I think that makes sense.

> I'm not seeing any duplicates in the site editor. Is that the post editor? I'm only changing the site editor here.

As I mentioned in the comment, it was a local made-up example showcasing that duplicates can exist in the `$paths` array. It was an example from the post editor.